source/RSRCFilter.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2020 Raheem Idowu <abdurraheemidowu@gmail.com>
+ * All rights reserved. Distributed under the terms of the MIT license.
+ */
+
+#include "RSRCFilter.h"
+
+#include <MimeType.h>
+#include <NodeInfo.h>
+#include <string.h>
+
+#include <compat/sys/stat.h>
+
+static const char* supported_types[] = {
+	"application/x-vnd.be-elfexecutable",
+	NULL
+};
+
+bool
+RSRCFilter::Filter(const entry_ref* ref, BNode* node, struct stat_beos* st,
+	const char* filetype)
+{
+	BEntry entry(ref, true);
+	if (entry.InitCheck() != B_OK)
+		return false;
+	if (entry.IsDirectory())
+		return true;
+
+	entry_ref entryRef;
+	if (entry.GetRef(&entryRef) != B_OK)
+		return false;
+
+	//Check if ref is a symlink
+	bool isLink = S_ISLNK(st->st_mode);
+	if (isLink) {
+		BNodeInfo * nodeInfo = new BNodeInfo(node);
+		char mimeType[B_MIME_TYPE_LENGTH];
+		if (nodeInfo->GetType(mimeType) != B_OK) {
+			delete nodeInfo;
+			return false;
+		} else {
+			delete nodeInfo;
+			filetype = mimeType;
+		}
+	}
+
+	//All files of non-BFS drives report as octet-stream or empty
+	if (strcasecmp(filetype, "application/octet-stream") == 0
+		|| filetype == "") {
```

Comment:
I don't think filetype == ""  will work

Reply:
It is supposed to be a check for an empty string, why would it not work?
What else could work?

Reply:
It won't work because `filetype` is a pointer and `""` is a string literal. How do you compare C strings? And more specifically, how do you check if a C string is a null/empty string?